clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/DaVinciRecordTransformer.java

```diff
@@ -1,26 +1,46 @@
 package com.linkedin.davinci.client;
 
+import com.linkedin.davinci.StoreBackend;
+import com.linkedin.davinci.store.AbstractStorageEngine;
 import com.linkedin.venice.annotation.Experimental;
+import com.linkedin.venice.blobtransfer.BlobTransferManager;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.serializer.AvroGenericDeserializer;
 import com.linkedin.venice.serializer.AvroSerializer;
+import com.linkedin.venice.utils.ComplementSet;
 import com.linkedin.venice.utils.lazy.Lazy;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
```

Comment:
So we're bringing in quite a lot of imports and logic to this interface.  Theres nothing technically wrong with this.

HOWEVER, I think it's distracting.  This interface is weather we like it or not, going to be a bit of user facing documentation.  To that end, let's see if we can slim down the amount of stuff the user has to look at when implementing this.  I think it'd be good if we had a protected/final/static object here that encapsulated the complicated logic a bit more.  That way this interfaces final methods just invoke object.recover/object.checkForChanges, and the user isn't confronted with all this stuff.